astropy/units/format/base.py

```diff
@@ -24,15 +26,96 @@ def __init_subclass__(cls, **kwargs):
         super().__init_subclass__(**kwargs)
 
     @classmethod
-    def parse(cls, s):
+    def _get_unit_name(cls, unit):
+        return unit.get_format_name(getattr(cls, "name", None))
+
+    @classmethod
+    def format_exponential_notation(cls, val, format_spec="g"):
         """
-        Convert a string to a unit object.
+        Formats a value in exponential notation.
+
+        Parameters
+        ----------
+        val : number
+            The value to be formatted
+
+        format_spec : str, optional
+            Format used to split up mantissa and exponent
+
+        Returns
+        -------
+        string : str
```

Comment:
```suggestion
        str
```

Reply:
Ha, didn't know that was allowed -- definitely makes sense. Adopted.